### DISCUSSION TITLE
Creating a grid like "graph paper" in typst

### DISCUSSION THREAD

**Discussion by CasparNuel** [typst/typst#4331] [Q&A] [answered]
Hi!

I am trying to create graph paper given a "box" size for the graph paper. In my example I am simply trying a 1CMx1CM grid pattern, but ideally it would be a parameter (trivial change).

I am currently struggling with getting this to work (not the parameter, but the grid graph paper).

My first idea was this, but it doesn't produce the desired result (only vertical lines, no horizontal lines):

```typ
#set page(margin: 0pt)

#repeat([#line(length: 100%, angle: 90deg)#h(1cm)])
#place([#line(length: 100%, angle: 0deg)#v(1cm)])
```

My second attempt is this, which produces output more like I want to achieve but still not ideal:

```typ
#set page(margin: 0pt)

#grid(columns: (1cm,) * 100, rows: (1cm,) * 100, stroke: (paint: black))
```

While this does create a 1CMx1CM grid graph paper-like output, it still has some shortcomings:

- The `* 100` for the rows causes overflow to the next page, which ideally would not happen
- Since an A4 paper is not exactly divisible in 1CM rows, the bottom row has a bit of white space underneath it. Ideally, the vertical lines would also go through this white space, preserving the grid-like visual over the entire page

Ideally, I would want to somehow detect exactly how many vertical and horizontal lines fit on a page (to avoid the current `* 100`) and only create a grid/pattern for that amount of space.

This could happen by taking (psuedocode) `columns=round(PAGE_WIDTH / 1CM); rows=round(PAGE_HEIGHT / 1CM)`.

I'm sure this is very easy in typst, I am just not creative enough to get a clean graph paper output with what I know currently :yum: . Any input to get this to work would be very appreciated!

**Comment by jamesrswift** [comment]
I imagine a pattern fill on a block with full width and height will do the trick nicely

**Reply by CasparNuel** [reply, marked as answer]
Thank you for your suggestion! After some research I landed on this approach, which does the job nicely. Thanks again!

```typ
#set page(margin: 0pt)

#let grid = pattern(size: (1cm, 1cm))[
  #place(line(start: (0%, 0%), end: (0%, 100%)))
  #place(line(start: (0%, 0%), end: (100%, 0%)))
]

#rect(fill: grid, width: 100%, height: 100%)
```
